core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -538,7 +538,10 @@ class PlaintextConsumerTest extends BaseConsumerTest {
   def testPositionAndCommit() {
     sendRecords(5)
 
-    assertNull(this.consumers.head.committed(new TopicPartition(topic, 15)))
+    // committed() on a partition that doesn't exist throws an exception
+    intercept[KafkaException] {
+      this.consumers.head.committed(new TopicPartition(topic, 15))
```

Comment:
Thanks for your detailed feedback on this.

The changes you suggested make sense. However, I'm not quite clear on what you mean by "updating internal offsets" in the first bullet, and would appreciate your clarification. We are going to exclude offset commit and fetch APIs here (as they are covered by the next bullet). Do we include here APIs like metadata update and fetch?

The 2nd bullet is already addressed by the current patch (as you also mentioned).

I also agree with your suggestion around topic deletion. I'll try to open a separate JIRA for that one.

Thanks again.